ext/msgpack/unpacker.c

```diff
@@ -168,12 +170,31 @@ static inline int object_complete_ext(msgpack_unpacker_t* uk, int ext_type, VALU
 
     VALUE proc = msgpack_unpacker_ext_registry_lookup(uk->ext_registry, ext_type);
     if(proc != Qnil) {
-        VALUE obj = rb_funcall(proc, s_call, 1, str);
-        return object_complete(uk, obj);
+        long arity = 0;
+        if (rb_obj_class(proc) == rb_cProc) {
+            arity = rb_proc_arity(proc);
+        } else if (rb_obj_class(proc) == rb_cMethod) {
+            arity = FIX2LONG(rb_funcall(proc, s_arity, 0));
+        }
```

Comment:
I really don't have having to check the arity before invoking procs, I think we should check it once when the extension is registered, and store it in the registry.

Reply:
> I really don't have having 
I meant don't like